examples/private_app_gateway/main.tf

```diff
@@ -0,0 +1,165 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+# This module generates the resource-name of resources based on resource_type, naming_prefix, env etc.
+module "resource_names" {
+  source = "git::https://github.com/launchbynttdata/tf-launch-module_library-resource_name.git?ref=1.0.0"
+
+  for_each = var.resource_names_map
+
+  region                  = join("", split("-", var.region))
+  class_env               = var.environment
+  cloud_resource_type     = each.value.name
+  instance_env            = var.environment_number
+  instance_resource       = var.resource_number
+  maximum_length          = each.value.max_length
+  logical_product_family  = var.logical_product_family
+  logical_product_service = var.logical_product_service
+}
+
+module "resource_group" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-resource_group.git?ref=1.0.0"
+
+  name     = local.resource_group_name
+  location = var.region
+  tags = {
+    resource_name = local.resource_group_name
+  }
+}
+
+module "network" {
+  source  = "Azure/vnet/azurerm"
+  version = "4.1.0"
+
+  resource_group_name = local.resource_group_name
+  use_for_each        = true
+  vnet_location       = var.location
+
+  address_space = var.address_space
+  subnet_names  = var.subnet_names
+
+  subnet_prefixes = var.subnet_prefixes
+  vnet_name       = local.vnet_name
+
+  depends_on = [module.resource_group]
+}
+
+module "network_security_group" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-network_security_group.git?ref=1.0.0"
+
+  name                = local.network_security_group_name
+  location            = var.location
+  resource_group_name = local.resource_group_name
+  security_rules      = local.all_security_rules
+
+  depends_on = [module.network]
+}
+
+module "nsg_subnet_association" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-nsg_subnet_association.git?ref=1.0.0"
+
+  network_security_group_id = module.network_security_group.network_security_group_id
+  subnet_id                 = local.app_gtwy_subnet_id
+
+  depends_on = [module.network_security_group]
+
+}
+
+module "app-gateway" {
```

Comment:
Any reason this is `kebab-case` where typically the module references we use are `snake_case`?

Reply:
Fixed this!!